Entity/CustomItem.php

```diff
@@ -611,4 +620,32 @@ public function getRelationsByType(string $entityType): Collection
                 return new ArrayCollection([]);
         }
     }
+
+    public function getUniqueHash(): ?string
+    {
+        return $this->uniqueHash;
+    }
+
+    public function setUniqueHash(?string $uniqueHash): void
+    {
+        $this->uniqueHash = $uniqueHash;
+    }
+
+    public function createUniqueHash(): ?string
+    {
+        $uniqueHash             = [];
+        $uniqueIdentifierFields = $this->customObject->getUniqueIdentifierFields();
+
+        if (0 === $uniqueIdentifierFields->count()) {
+            return null;
+        }
+
+        foreach ($uniqueIdentifierFields->getValues() as $uniqueIdentifierField) {
+            $uniqueIdentifierFieldAlias              = $uniqueIdentifierField->getAlias();
+            $uniqueHash[$uniqueIdentifierFieldAlias] = $this->findCustomFieldValueForFieldAlias($uniqueIdentifierFieldAlias)->getValue();
+        }
+        ksort($uniqueHash); //sort array on the basis of key so that the order of keys is the same everytime
```

Comment:
I don't think comments that explain php functions add any value. If someone doesn't know what it does they can check through their IDE or PHP Documentation.

If we need a comment here, I think it should explain why we need to preserve the key order instead.

Reply:
Very well. I'll modify the comment for a better worded explanation. Thanks!